src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -1207,8 +1207,9 @@ void GrpcLb::FillChildRefsForChannelz(
     channelz::ChildRefsList* child_subchannels,
     channelz::ChildRefsList* child_channels) {
   // delegate to the RoundRobin to fill the children subchannels.
-  rr_policy_->FillChildRefsForChannelz(child_subchannels, child_channels);
-  MutexLock lock(&lb_channel_mu_);
```

Comment:
Why is it unnecessary to take this lock here?  It looks to me like we do need to do so to ensure that `lb_channel_` is not in the process of being destroyed when we get here (see the code in `ShutdownLocked()`).

Conversely, if there is a reason why we don't actually need this lock here, then the lock is useless, since it's not being used for anything else.  In that case, we should remove it completely.